pkg/driver/camera/camera_linux.go

```diff
@@ -92,21 +101,21 @@ func (c *camera) VideoRecord(p prop.Media) (video.Reader, error) {
 
 	r := video.ReaderFunc(func() (img image.Image, err error) {
 		// Wait until a frame is ready
-		for {
+		for i := 0; i < maxErrorCount; i++ {
 			err := c.cam.WaitForFrame(5)
 			switch err.(type) {
 			case nil:
 			case *webcam.Timeout:
-				continue
+				return nil, errReadTimeout
 			default:
 				// Camera has been stopped.
-				return nil, io.EOF
+				return nil, err
```

Comment:
Sadly, we somehow need to return `io.EOF` because this is a way for the reader consumer to know when a reader is done emitting values. Essentially, I'm trying to follow `io.Reader` rules.

Reply:
In my understanding, `io.Reader` returns `io.EOF` only if reaches the end of the data and has no error, and consumer can't assume that the reader resumes after any error.
Please let me know if my understanding is wrong.

Reply:
Yep, that's definitely true. I think my wording was just really bad. `io.EOF` is definitely like any error, except it adds value to the caller that this is one of the error values which they can ignore (since it's not really an error but rather just a way to tell the caller that it's the end of file).

I guess the difference is that if there's another function that takes this reader, it can ignore `io.EOF` and still return `nil` for its own error.

A good example is `io.Copy` which under the hood calls `copyBuffer`, https://golang.org/src/io/io.go?s=13434:13516#L380. In there, you can see that it treats `io.EOF` as non-error.

Reply:
Yes, but I think it's better to notify device disconnection or IO failure kind of errors as error.
I digged into webcam package now, and found that `WaitForFrame` and `ReadFrame` only returns error fom `unix.Select` and `unix.Syscall`, they basically a kind of fatal.

I think it would be better to return `io.EOF` if `Read` is called after `Close`.

Reply:
> I think it would be better to return `io.EOF` if `Read` is called after `Close`.

I'll add this.